recipe/build.sh

```diff
@@ -4,9 +4,17 @@ set -exuo pipefail
 
 mkdir -p build-target
 
+XGB_CMAKE_ARGS=()
+
+if [[ "$target_platform" == osx-* ]]
+then
+    XGB_CMAKE_ARGS=(-DOpenMP_C_FLAGS="-Xpreprocessor -fopenmp -I${PREFIX}/include" -DOpenMP_CXX_FLAGS="-Xpreprocessor -fopenmp -I${PREFIX}/include" -DOpenMP_C_LIB_NAMES=libomp -DOpenMP_CXX_LIB_NAMES=libomp -DOpenMP_libomp_LIBRARY=${PREFIX}/lib/libomp.dylib ${XGB_CMAKE_ARGS[@]+"${XGB_CMAKE_ARGS[@]}"} )
+fi
+
+
 if [[ ${cuda_compiler_version} != "None" ]]; then
-    export CMAKE_ARGS="-DUSE_CUDA=ON -DUSE_NCCL=ON -DBUILD_WITH_SHARED_NCCL=ON ${CMAKE_ARGS}"
-    export CMAKE_ARGS="-DBUILD_WITH_CUDA_CUB=ON -DPLUGIN_RMM=ON ${CMAKE_ARGS}"
+    XGB_CMAKE_ARGS=(-DUSE_CUDA=ON -DUSE_NCCL=ON -DBUILD_WITH_SHARED_NCCL=ON ${XGB_CMAKE_ARGS[@]+"${XGB_CMAKE_ARGS[@]}"} )
+    XGB_CMAKE_ARGS=(-DBUILD_WITH_CUDA_CUB=ON -DPLUGIN_RMM=ON ${XGB_CMAKE_ARGS[@]+"${XGB_CMAKE_ARGS[@]}"} )
 fi
```

Comment:
Could someone please verify we are passing the right flags here? Also please feel free to update if not